### ISSUE TITLE
Plotting problems with scan_nd

### ISSUE THREAD

**Issue by cmazzoli** [NSLS-II/bluesky#868]
Asked a scan in 2 motors, got it but figure prepared for getting a mesh in the same 2 motors instead
For details, see:
https://logbook.nsls2.bnl.gov/23-ID-1/#63611_1

````
In [79]: th2th_pos = cycler(theta, thetas) + cycler(delta, deltas)
In [81]: dets = [fccd]

In [82]: RE(mv(fccd.exposure, (1, 0, 1)))
Out[82]: ()
...

In [85]: RE(scan_nd(dets,th2th_pos))
Transient Scan ID: 86050     Time: 2017/10/22 20:17:40
Persistent Unique Scan ID: 'aab28779-4e4f-473c-8c06-2f9bd6d4fc32'
Transient Scan ID: 86050 @ 2017/10/22 20:17:40
Persistent Unique Scan ID: 'aab28779-4e4f-473c-8c06-2f9bd6d4fc32'
New stream: 'baseline'
New stream: 'primary'                                                                                                                                                         
+-----------+------------+------------+------------+-------------------+-------------------+                                                                                  
|   seq_num |       time |      delta |      theta | fccd_stats3_total | fccd_stats4_total |
+-----------+------------+------------+------------+-------------------+-------------------+
|         1 | 20:17:47.8 |    40.0000 |    19.8700 |         -16613114 |        -259419609 |
|         2 | 20:18:05.9 |    41.0000 |    20.3700 |            408778 |           5242892 |                                                                                  
|         3 | 20:18:10.8 |    42.0000 |    20.8700 |            334858 |           3953314 |                                                                                  
/home/xf23id1/conda_envs/dev/lib/python3.6/site-packages/ophyd/status.py:336: RuntimeWarning: divide by zero encountered in double_scalars                                    
  time_remaining = time_elapsed / fraction + self.settle_time
|         4 | 20:18:15.9 |    43.0000 |    21.3700 |            267732 |           3321314 |                                                                                  
|         5 | 20:18:21.7 |    44.0000 |    21.8700 |            201152 |           3414689 |                                                                                  
|         6 | 20:18:26.7 |    45.0000 |    22.3700 |            126287 |           3231229 |                                                                                  
|         7 | 20:18:35.3 |    45.9965 |    22.8700 |            105760 |           4082372 |                                                                                  
|         8 | 20:18:40.4 |    47.0000 |    23.3700 |             61969 |           3338174 |                                                                                  
|         9 | 20:18:45.4 |    48.0000 |    23.8700 |             65041 |           3456289 |                                                                                  
|        10 | 20:18:50.4 |    49.0000 |    24.3700 |            100508 |           3534242 |                                                                                  
|        11 | 20:18:56.1 |    50.0000 |    24.8700 |            161385 |           3405193 |                                                                                  
|        12 | 20:19:01.1 |    51.0000 |    25.3700 |            184970 |           3067978 |                                                                                  
|        13 | 20:19:06.8 |    52.0000 |    25.8700 |            160322 |           2457811 |                                                                                  
|        14 | 20:19:11.8 |    53.0000 |    26.3700 |            115579 |           2160564 |                                                                                  
|        15 | 20:19:16.8 |    54.0000 |    26.8700 |             70986 |           1673438 |                                                                                  
|        16 | 20:19:21.7 |    55.0000 |    27.3700 |             33682 |           1414459 |                                                                                  
|        17 | 20:19:26.6 |    56.0000 |    27.8700 |              6921 |           1446007 |                                                                                  
|        18 | 20:19:31.6 |    57.0000 |    28.3700 |              7480 |            938767 |                                                                                  
|        19 | 20:19:37.2 |    58.0000 |    28.8700 |              9786 |            873456 |                                                                                  
|        20 | 20:19:42.2 |    59.0000 |    29.3700 |              5404 |            477971 |                                                                                  
+-----------+------------+------------+------------+-------------------+-------------------+                                                                                  
generator scan_nd ['aab287'] (scan num: 86050)
Out[85]: ('aab28779-4e4f-473c-8c06-2f9bd6d4fc32',)

````

## Steps to Reproduce (for bugs)

1.
2.
3.

**Comment by jrmlhermitte** [comment]
addressed with #917 can someone maybe test this if they have time? :-)

**Comment by tacaswell** [comment]
@ambarb is happy that this be closed as you can do this scan with `inner_product_scan` and she is happy with the plots.
